pkg/topology/depthmonitor/depthmonitor.go

```diff
@@ -96,27 +89,21 @@ func (s *Service) manage(warmupTime time.Duration) {
 	case <-time.After(warmupTime):
 	}
 
-	// check if we have already saved a depth value before shutting down previously.
-	var initialDepth uint8
-	err := s.st.Get(depthKey, &initialDepth)
-	if err != nil && !errors.Is(err, storage.ErrNotFound) {
-		s.logger.Errorf("depthmonitor: failed reading stored depth: %s", err.Error())
-	}
+	// wire up batchstore to start reporting storage radius to kademlia
+	s.bs.SetStorageRadiusSetter(topology)
```

Comment:
Although this makes sense, I would prefer not doing this here. This should be part of batchstore initialization in node.go and not depthmonitor. In case of light nodes we could initialize it with a dummy setter which does nothing.

This can be easily missed if we put it here.

Reply:
so storage radius is zero initially. Because right under this line we change the storage radius in the case that it's zero, this is the exact place we need to wire kademlia to batchstore.

Reply:
a zero storage radius must not be reported to kademlia, in short